### PR TITLE
python3Packages.cgal: init at unstable-20220516

### PR DESCRIPTION
**Pull request by JCoMcL**
###### Description of changes
Package [CGAL](https://github.com/CGAL/cgal) SWIG bindings. The project also supports Java bindings though I've only packaged it for Python.
Let me know if the pakcage name ought to change, it could be any combination of:
- cgal_5
- cgal-python
- cgal-bindings
I just defaulted to the most succinct one

###### Things done

- Built on `x86_64-linux`
- Tested that module could be imported
- Tested some [example code](https://github.com/CGAL/cgal-swig-bindings/blob/main/examples/python/polygonal_triangulation.py)
- Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be:
```
python3Packages.cgal: init at <version>
```

IIRC, we are stilling doing `unstable_YYYYMMDD` for commit pinned commits

**Comment by dotlambda** [comment]
> IIRC, we are stilling doing `unstable_YYYYMMDD` for commit pinned commits

`unstable_YYYYMMDD` or `unstable-YYYYMMDD`?

**Comment by JCoMcL** [comment]
The upstream doesn't seem to have any version numbering system, nor does it seem to have a relationship with any specifc version of CGAL.
> IIRC, we are stilling doing unstable_YYYYMMDD for commit pinned commits
I had to read this twice but I think I get you.

I started a discourse thread a while ago about my woes trying to package this, I'll probably put an addendum to it once this is approved.

File: pkgs/development/python-modules/cgal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/cgal/default.nix:34]:
```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/development/python-modules/cgal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/cgal/default.nix:35]:
```suggestion
```
not needed for Python packages.

File: pkgs/development/python-modules/cgal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/cgal/default.nix:13]:
```suggestion
  version = "unstable-20220516";
```

File: pkgs/development/python-modules/cgal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/cgal/default.nix:3]:
```suggestion
, python
```
because we want different packages for Python 3.9 and Python 3.10

File: pkgs/development/python-modules/cgal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/cgal/default.nix:22]:
```suggestion
  buildInputs = [ cgal_5 python ] ++ cgal_5.buildInputs;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please add yourself to `maintainers/maintainer-list.nix` in a separate commit.
Also, don't forget to add a line to `pkgs/top-level/python-packages.nix`.

**Comment by dotlambda** [comment]
> * Tested that module could be imported
> * Tested some [example code](https://github.com/CGAL/cgal-swig-bindings/blob/main/examples/python/polygonal_triangulation.py)

How exactly did you test these?

**Comment by JCoMcL** [comment]
> How exactly did you test these?

 I did `nixos-rebuild test -I nixpkgs=<path to your local nixpkgs folder>` as described in [the manual](https://nixos.org/manual/nixpkgs/stable/#chap-submitting-changes), verified that `python.withPackages` worked with my package, verified that I could import it (`python -c 'import CGAL'`) and that I could use it without it crashing. If it's automated testing you want, I haven't done anything of the sort.
 
 > Please add yourself to maintainers/maintainer-list.nix in a separate commit.
 
 Mind telling me what I'm signing myself up for by making myself the maintainer of this package? If it's NixOS policy to not accept packages before someone's promised to make sure it stays up-to-date then perhaps this package isn't a good fit. I'm only submitting it because I had it and thought I might-as-well share. I don't mean to make a fuss, I just want to by sure I'm not making any promises I can't keep.

**Comment by dotlambda** [comment]
@JCoMcL It's up to your interpretation. You might be pinged if there's a problem with the package or in a PR that modifies it. It would be nice if you kept it up to date but we can't force you to.

**Comment by JCoMcL** [comment]
Fair enough, I think that's everything

File: pkgs/development/python-modules/cgal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/cgal/default.nix:18]: `unstable-20220516` is not a tag

File: maintainers/maintainer-list.nix
Comment (dotlambda) [inline maintainers/maintainer-list.nix:5754]: `github` and `githubId` are missing

**Review by dotlambda — CHANGES_REQUESTED** [review]
The first commit should only touch `maintainers/maintainer-list.nix` and have the commit message `maintainers: add jcomcl`.

**Pull request closed without merging**
